src/schedule/edf_schedule.c

```diff
@@ -201,6 +239,12 @@ static int schedule_edf_task_running(void *data, struct task *task)
 	task_context_set(edf_pdata->ctx);
 	task->state = SOF_TASK_STATE_RUNNING;
 
+	if (edf_pdata->budget) {
```

Comment:
Can you comment what this is doing and also the choice of timers, should we use the CAVS timer ?